redis/src/cluster.rs

```diff
@@ -232,7 +232,7 @@ where
             connections: RefCell::new(HashMap::new()),
             slots: RefCell::new(SlotMap::new(cluster_params.read_from_replicas)),
             auto_reconnect: RefCell::new(true),
-            read_timeout: RefCell::new(None),
+            read_timeout: RefCell::new(Some(cluster_params.response_timeout)),
```

Comment:
Is this needed for this PR? If not can we move to a separate commit?

Reply:
I'll split the commits. This was added because the split build raised warnings about unused fields.

Reply:
done. in the long term, we should probably use https://github.com/TheBevyFlock/flag-frenzy, https://crates.io/crates/cargo-hack, or https://github.com/frewsxcv/cargo-all-features to properly test this.

Reply:
Ah, found it :laughing: Should this be re-based on the other PR?

Reply:
actually, this PR can be closed in favor of the other PR, since it gives a more through check.